### PR TITLE
feat!: plat-6785 upgrade cdk 1.x -> 2.x

### PR DESCRIPTION
**Pull request by malcyL**
- https://github.com/talis/platform/issues/6785

This PR Upgrades the construct library from CDK 1.x to 2.x.

This is a breaking change which will require all consumers of the library to upgrade to CDK 2.x before they can upgrade to this version of the library.

Migrations guides:

- For the library itself: https://docs.aws.amazon.com/cdk/v2/guide/migrating-v2.html
- For the changes to testing:  https://github.com/aws/aws-cdk/blob/v1-main/packages/@aws-cdk/assertions/MIGRATING.md

As part of this story, to be able to deploy the sample stacks for development and to run integration tests against, the shared aspire account had to be re bootstrap with the command:

```
cdk bootstrap aws://302477901552/eu-west-1
```

This will need to be done in the devops account, for the appropriate regions, as part of the deployment of services using this library.

After compilation problems trying to upgrade withing the existing cdk 1.x project, the approach taken by this PR was to create an empty 2.x project, and then reinstate all the code, tests and example projects gradually, fixing issues along the way.

### CODE REVIEW

File: .circleci/config.yml
Comment (malcyL) [inline .circleci/config.yml:66]: Remove before merge.

**Review by kiyanwang — APPROVED** [review]
I think this looks ok, there's a lot of boilerplate changes, which we would expect as well as import renaming etc. 

The only thing that made me stop and think was the issue with lambdaSecurityGroups in the examples, I understand why you've commented those out, I wonder if there's anything else we should be doing, but I'll defer to Grzeg or Stephen's opinion.

File: .eslintrc.js
Comment (malcyL) [reply .eslintrc.js:6]: This is prolific through the code:

```bash
ml@kaylee:~/dev/talis-cdk-constructs$ npm run lint                                                                                                                                                         [1/1856]

> talis-cdk-constructs@0.1.0 lint     
> npm run lint:prettier && npm run lint:eslint


> talis-cdk-constructs@0.1.0 lint:prettier
> prettier --check .

Checking formatting...
All matched files use Prettier code style!

> talis-cdk-constructs@0.1.0 lint:eslint
> eslint . --ext .ts


/home/ml/dev/talis-cdk-constructs/lib/aspects/resource_prefixer/resource_prefixer.ts
  19:46  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  22:17  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ml/dev/talis-cdk-constructs/src/lambda/api/PersonaAuthorizer.d.ts
  13:12  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  14:14  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  16:24  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  16:38  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  17:23  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  18:35  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  18:64  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  42:43  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ml/dev/talis-cdk-constructs/src/lambda/api/PersonaAuthorizer.ts
   52:10  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   53:12  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   56:22  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   56:36  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  111:37  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  111:64  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  134:33  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  134:62  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  139:17  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  139:26  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  139:45  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ml/dev/talis-cdk-constructs/src/lambda/api/authorizer.ts
  3:46  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  3:60  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/ml/dev/talis-cdk-constructs/src/types/talis-node/index.d.ts
  11:25  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  11:40  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  11:46  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  15:28  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  15:49  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  17:25  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  17:38  error  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 30 problems (30 errors, 0 warnings)
```

I'm not sure why this wasn't erroring before - but we have bumped versions of things.

I don't think we should get into changing the code as part of this PR.

Adding this here seemed preferable to adding excludes everywhere in the code.

File: examples/simple-lambda-worker/jest.config.js
Comment (malcyL) [reply examples/simple-lambda-worker/jest.config.js:4]: The approach taken here, with the main library and all the individual examples, was to nuke the projects, create a new cdk v2 empty project and re add the necessary files.

This file is generated from the `cdk init` command. I've just tested craeting an empty project, and it creates them with `'` not `"`.

So - that regeneration of the project is how this change was made.

I guess this points out that we are not running prettier over the example projects though.

File: lib/authenticated-api/authenticated-api-function-props.ts
Comment (malcyL) [reply lib/authenticated-api/authenticated-api-function-props.ts:12]: I added comments in the examples - but not here.

The vpc has always been optional. The examples were setting an existing security group, but not a vpc. Doing this in cdk v2 results in the error:

```
Error: Cannot configure 'securityGroups' without configuring a VPC
```

So - if the vpc is optional, the securityGroups has to be optional too. I modified the examples to not set a securityGroup.

This change is also what Nad pointed out above:

image

File: .eslintrc.js
Comment (mrliptontea) [reply .eslintrc.js:6]: You can disable the rule for entire file - I'd be happier with that, and it's only five of them.
https://eslint.org/docs/latest/use/configure/rules#disabling-rules

And looking at the list of files it makes sense to disable the rule in those, but we don't want to allow `any` elsewhere!

**Review by mrliptontea — APPROVED** [review]

**Pull request merged**
